### PR TITLE
Implement Ore ID account provisioning flow

### PR DESCRIPTION
**Pull request by Alexey1100**
Resolves:
https://www.pivotaltracker.com/story/show/175423514
https://www.pivotaltracker.com/story/show/175214214
https://www.pivotaltracker.com/story/show/175601638
https://www.pivotaltracker.com/story/show/175943963

- Define auto_provisioning_stage enum
- Define provisioning jobs and helpers
- Add custodial tx creation to OreIdService
- Store OreID temp_password on custodial account creation
- Add a message regarding OreId wallet sync

### CODE REVIEW

File: app/models/ore_id_account.rb [hunk 1]
```diff
@@ -31,13 +58,56 @@ def sync_wallets
     ok!
   end
 
+  def sync_balance
+    balance = Comakery::Algorand(wallets.last.blockchain).account_balance(wallets.last.address)
```
Comment (Alexey1100) [inline app/models/ore_id_account.rb:62]: Better to be extracted into Algorand subclass and called from Wallet model:

`wallet.blockchain.balance`

File: app/models/ore_id_account.rb [hunk 2]
```diff
@@ -31,13 +58,54 @@ def sync_wallets
...
+  def sync_balance
+    if wallets.last.instant_balance.positive?
+      initial_balance_confirmed!
+    else
+      raise StandardError, 'Account balance is 0'
+    end
+  end
+
+  def create_opt_in_tx
+    # Create TX
```
Comment (Alexey1100) [inline app/models/ore_id_account.rb:70]: #1748

File: app/controllers/auth/ore_id_controller.rb [hunk 3]
```diff
@@ -20,7 +20,7 @@ def receive
     verify_received_account or return
 
     if current_ore_id_account.update(account_name: params.require(:account), state: :ok)
-      flash[:notice] = 'ORE ID Linked'
+      flash[:notice] = 'ORE ID Linked. Syncrhonising wallets...'
```
Comment (Riddlerrr) [inline app/controllers/auth/ore_id_controller.rb:23]: Typo here `Synchronising`

**Review by Riddlerrr — APPROVED** [review]

File: app/controllers/auth/ore_id_controller.rb (hunk 3 above)
Comment (Alexey1100) [inline app/controllers/auth/ore_id_controller.rb:23]:
```suggestion
      flash[:notice] = 'ORE ID Linked. Synchronising wallets...'
```

**Review by aquabu — APPROVED** [review]

**Review by aquabu — APPROVED** [review]

**Pull request merged**